hack/build-manifests.sh

```diff
@@ -1,113 +1,214 @@
 #!/usr/bin/env bash
 set -e
 
-# TODO: If we create more hack scripts this should go in common
-# and be sourced
+# build-manifests is designed to populate the deploy directory
+# with all of the manifests necessary for use in development
+# and for consumption with the operator-lifecycle-manager.
+# 
+# First, we create a temporary directory and filling it with
+# all of the component operator's ClusterServiceVersion (CSV for OLM)
+# and CustomResourceDefinitions (CRDs); being sure to copy the CRDs
+# into the deploy/crds directory.
+# 
+# The CSV manifests contain all of the information we need to 1) generate
+# a combined CSV and 2) other development related manifests (like the
+# operator deployment + rbac).
+#
+# Second, we pass all of the component CSVs off to the manifest-templator
+# that handles the deployment specs, service account names, permissions, and
+# clusterPermissions by converting them into their corresponding Kubernetes
+# manifests (ie. permissions + serviceAccountName = role + service account
+# + role binding) before writing them to disk.
+#
+# Lastly, we take give the component CSVs to the csv-merger that combines all
```

Comment:
Thanks for adding this David, very helpful. One question, why does build-manifest.sh create a _merger file now? But in 0.0.1/0.0.2/0.0.3 directories under deploy/olm-catalog/kubevirt-hyperverged there isn't one? Is there an extra step one needs to do to create a new version?

Reply:
I left the `templates/` directory alone for this Pull Request to keep the size down but that should be removed in a follow-up.

Previously, the `templates/olm-catalog/kubevirt-hyperconverged/VERSION/kubevirt-hyperconverged-operator.VERSION.clusterserviceversion_merger.yaml.in` was taken and put into the container image (downstream) for use with merging the CSVs together.

Now, we are just putting all of the HCO related objects in source and building the CSV from the `components` package.

To create a new version, you would just change the `CSV_VERSION` (and probably also the `REPLACES_VERSION`) and run `hack/build-manifests.sh`. The directory should be created for you.